src/test/java/org/broadinstitute/hellbender/tools/reference/CompareReferencesIntegrationTest.java

```diff
@@ -118,19 +119,109 @@ public void testCompareReferencesMultipleReferencesStdOut() throws IOException{
+        final File actualOutput = new File(output, "hg19mini.fasta_hg19mini_chr1indel.fasta.vcf");
+        IntegrationTestSpec.assertEqualTextFiles(actualOutput, expectedOutput);
+    }
+
+    @Test
+    public void testFullAlignmentModeInsertion() throws IOException{
+        final File ref1 = new File(getToolTestDataDir() + "hg19mini_chr1indel.fasta");
+        final File ref2 = new File(getToolTestDataDir() + "hg19mini.fasta");
+        final File output = IOUtils.createTempDir("tempFullAlignmentIndel");
+
+        final String[] args = new String[] {"-R", ref1.getAbsolutePath() , "-refcomp", ref2.getAbsolutePath(), "-base-comparison", "FULL_ALIGNMENT", "-base-comparison-output", output.getAbsolutePath()};
+        runCommandLine(args);
+
+        final File expectedOutput = new File(getToolTestDataDir(), "expected.testInsertion.hg19mini_chr1indel.fasta_hg19mini.fasta.vcf");
+        final File actualOutput = new File(output, "hg19mini_chr1indel.fasta_hg19mini.fasta.vcf");
+        IntegrationTestSpec.assertEqualTextFiles(actualOutput, expectedOutput);
+    }
+
+    @Test
+    public void testFullAlignmentSNPsOnMultipleContigs() throws IOException{
+        final File ref1 = new File(getToolTestDataDir() + "hg19mini.fasta");
+        final File ref2 = new File(getToolTestDataDir() + "hg19mini_snpsmultiplecontigs.fasta");
+        final File output = IOUtils.createTempDir("tempFullAlignmentSNPsMultipleContigs");
+
+        final String[] args = new String[] {"-R", ref1.getAbsolutePath() , "-refcomp", ref2.getAbsolutePath(), "-base-comparison", "FULL_ALIGNMENT", "-base-comparison-output", output.getAbsolutePath()};
+        runCommandLine(args);
+
+        final File expectedOutput = new File(getToolTestDataDir(), "expected.hg19mini.fasta_hg19mini_snpsmultiplecontigs.fasta.vcf");
+        final File actualOutput = new File(output, "hg19mini.fasta_hg19mini_snpsmultiplecontigs.fasta.vcf");
+        IntegrationTestSpec.assertEqualTextFiles(actualOutput, expectedOutput);
+    }
+
+    @Test
+    public void testFullAlignmentModeSNPAndINDEL() throws IOException{
+        final File ref1 = new File(getToolTestDataDir() + "hg19mini.fasta");
+        final File ref2 = new File(getToolTestDataDir() + "hg19mini_snpandindel.fasta");
+        final File output = IOUtils.createTempDir("tempFullAlignmentIndel");
+
+        final String[] args = new String[] {"-R", ref1.getAbsolutePath() , "-refcomp", ref2.getAbsolutePath(), "-base-comparison", "FULL_ALIGNMENT", "-base-comparison-output", output.getAbsolutePath()};
+        runCommandLine(args);
+
+        final File expectedOutput = new File(getToolTestDataDir(), "expected.SNPandINDEL.hg19mini.fasta_hg19mini_snpandindel.fasta.vcf");
+        final File actualOutput = new File(output, "hg19mini.fasta_hg19mini_snpandindel.fasta.vcf");
+        IntegrationTestSpec.assertEqualTextFiles(actualOutput, expectedOutput);
+    }
 }
```

Comment:
We ideally want a FULL_ALIGNMENT mode test case that comprehensively exercises all of the state transitions in the VCF creation code between SNPs, insertions, and deletions. Eg., something like this:

```
SNP
SNP
insertion
SNP
deletion
SNP
insertion
insertion
deletion
deletion
insertion
SNP
```

Is it possible to create a test case with all of the above on a single contig (and in that order)?